pkg/action/executor/gitops/github_manager.go

```diff
@@ -464,6 +539,26 @@ func (g *GitHandler) newPR(resName string, newBranch string) (*github.PullReques
 	return pr, nil
 }
 
+func (g *GitHandler) checkPrExists(resName, resNamespace, clusterId string) (*github.PullRequest, error) {
+	options := &github.PullRequestListOptions{
+		Base: g.baseBranch,
+		Head: g.commitUser, // ex head=turbonomic or head=turbonomic: or head=turbonomic:<branch-name>
+	}
+	prList, _, err := g.client.PullRequests.List(g.ctx, g.user, g.repo, options)
```

Comment:
nit: You may want to pull the open PRs only.
https://docs.github.com/en/rest/pulls/pulls#list-pull-requests